Microprofile 5.0 Pre-Release (#226)
eclipse/microprofile: spec/src/main/asciidoc/architecture.asciidoc

In reverse chronological order, here's the specification for each MicroProfile release.

> [[microprofile5.0-RC1]]

Reviewer: Per the following comments, I think it's best to have a look at the definition of the 4.1 release in this file. It has been updated a lot compared to 4.0.

Author: I think in addition to your recommendation about having a look at 4.1 definition, maybe this link could summarize the most important about  5.0: 

https://projects.eclipse.org/projects/technology.microprofile/releases/5.0

what do you think ?

Reviewer: Correct, this is the link to the release plan that should contain all necessary details of the release

> === MicroProfile 5.0 Release Candidate 1 (Q32021)

Reviewer: From MP 4.1 on, we decided to drop the quarter notion in this file. So this Q32021 notion can be removed

Author: Done

> // MicroProfile 5.0-RC1 is the 16th platform release for the

Reviewer: In 4.1, we decided to drop the number of platform releases we have done. So this "16th" notion can be removed

Author: line is commented, anyway I'm removing it!!

* https://github.com/eclipse/microprofile-health/releases/tag/4.0-RC2[Health TODO]
* https://github.com/eclipse/microprofile-metrics/releases/tag/4.0-RC1[Metrics TODO]
* https://download.eclipse.org/microprofile/staging/microprofile-open-api-3.0-RC1/microprofile-openapi-spec-3.0-RC1.html#_incompatible_changes[Open API]
* https://download.eclipse.org/microprofile/microprofile-rest-client-3.0-RC2/microprofile-rest-client-spec-3.0-RC2.html[Rest Client - BROKEN LINK]
* https://github.com/eclipse/microprofile-jwt-auth/releases/tag/2.0-RC2[JWT TODO]
* https://download.eclipse.org/microprofile/staging/microprofile-opentracing-3.0-RC2/microprofile-opentracing-spec-3.0-RC2.html#_incompatible_changes[Open Tracing]

If you are still dependent on Java EE 8, please consider using the https://github.com/eclipse/microprofile/releases/tag/4.1[4.1 release of MicroProfile].
If you are still dependent on Java EE 7, please consider using the https://github.com/eclipse/microprofile/releases/tag/1.4[1.4 release of MicroProfile].

The complete list of MicroProfile 5.0-RC1 specifications includes:

> 

Reviewer: The versions between brackets need to match with the links

Author: It matches with the link, does not it? are you referring to the line [56] recommending MP1.4 if you still using Java EE 7?

Reviewer: I think these above are fine, but these below are not aligned yet. E.g.: /tag/3.0-RC4[MicroProfile Config 2.0]

Author: Ohhhhh I see what you meant, these are fixed now!!! thanks

In reverse chronological order, here's the specification for each MicroProfile release.

[[microprofile5.0-RC1]]
=== MicroProfile 5.0 Release Candidate 1

> MicroProfile 5.0-RC1 is aligning every specification with Jakarta EE 9.1. This effectively creates an incompatible change due to the namespace change from javax to jakarta in the code.

Reviewer 2: ```suggestion
MicroProfile 5.0-RC1 aligns with Jakarta EE 9.1. It means this release will not work with previous version of Jakarta EE such as Jakarta EE 8. This effectively creates an incompatible change due to the namespace change from javax to jakarta in the code. 
```

Author: Done!! Thanks

    Config 3.0
    Fault Tolerance 4.0
    Health 4.0
    JWT RBAC 2.0
    Metrics 4.0
    Open API 3.0
    Open Tracing 3.0
    Rest Client 3.0

The following specifications include API incompatible changes:

> * https://download.eclipse.org/microprofile/staging/microprofile-config-3.0-RC4/microprofile-config-spec-3.0-RC4.html#_incompatible_changes[Config]

Reviewer 2: please use 3.0-RC5

> * https://download.eclipse.org/microprofile/staging/microprofile-config-3.0-RC5/microprofile-config-spec-3.0-RC5.html#_incompatible_changes[Config]

Reviewer 2: ```suggestion
* https://download.eclipse.org/microprofile/microprofile-config-3.0-RC5/microprofile-config-spec-3.0-RC5.html#_incompatible_changes[Config]
```

Author: Fixed!!! thx

> * https://download.eclipse.org/microprofile/microprofile-fault-tolerance-4.0-RC3/microprofile-fault-tolerance-spec-4.0-RC3.html#_incompatible_changes[Fault Tolerance]

Reviewer 2: ```suggestion
* https://download.eclipse.org/microprofile/microprofile-fault-tolerance-4.0-RC4/microprofile-fault-tolerance-spec-4.0-RC4.html#_incompatible_changes[Fault Tolerance]
```

Author: Fixed

> * https://download.eclipse.org/microprofile/staging/microprofile-fault-tolerance-4.0-RC2/microprofile-fault-tolerance-spec-4.0-RC2.html#_incompatible_changes[Fault Tolerance]

Reviewer 3: these should be moved out of /staging directory before we push 5.0-RC1

Reviewer: @emecas let's monitor these staging links. When the links without the "staging" part is available, the links can be updated here

Reviewer 2: I'll release more update for the specs in the next couple of days and will push the latest out of staging.

> * https://github.com/eclipse/microprofile-health/releases/tag/4.0-RC2[Health TODO]

Author: Updated!!

> * https://download.eclipse.org/microprofile/microprofile-metrics-4.0-RC2/microprofile-metrics-spec-4.0-RC2.html#_breaking_changes[Metrics --> This section SHOULD BE incompatible changes ]

Reviewer 2: ```suggestion
* https://download.eclipse.org/microprofile/microprofile-metrics-4.0-RC4/microprofile-metrics-spec-4.0-RC4.html#_incompatible_changes[Metrics]
```

Author: Fixed

> * https://download.eclipse.org/microprofile/microprofile-open-api-3.0-RC2/microprofile-openapi-spec-3.0-RC2.html#_incompatible_changes[Open API]

Reviewer 2: ```suggestion
* https://download.eclipse.org/microprofile/microprofile-open-api-3.0-RC3/microprofile-openapi-spec-3.0-RC3.html#_incompatible_changes[Open API]
```

Author: Thanks, I've updated  open-api to RC3 in 3 more places
